pkg/ring/consul_client.go

```diff
@@ -29,12 +30,15 @@ type ConsulConfig struct {
 }
 
 // RegisterFlags adds the flags required to config this to the given FlagSet
-func (cfg *ConsulConfig) RegisterFlags(f *flag.FlagSet) {
-	f.StringVar(&cfg.Host, "consul.hostname", "localhost:8500", "Hostname and port of Consul.")
-	f.StringVar(&cfg.Prefix, "consul.prefix", "collectors/", "Prefix for keys in Consul.")
-	f.StringVar(&cfg.ACLToken, "consul.acltoken", "", "ACL Token used to interact with Consul.")
-	f.DurationVar(&cfg.HTTPClientTimeout, "consul.client-timeout", 2*longPollDuration, "HTTP timeout when talking to consul")
-	f.BoolVar(&cfg.ConsistentReads, "consul.consistent-reads", true, "Enable consistent reads to consul.")
+func (cfg *ConsulConfig) RegisterFlags(f *flag.FlagSet, prefix string) {
+	if prefix != "" {
+		prefix = prefix + "."
+	}
```

Comment:
Same as on `KVClientConfig`, lets not do this and instead ask callers to append a dot.